src/database/utility.cpp

```diff
@@ -1,7 +1,118 @@
+#include <stdio.h>
+#include <iostream>
+#include <vector>
+#include <fstream>
 #include "utility.h"
+#include "field.h"
+#include "heap_page.h"
 
 namespace buzzdb {
-/**
- * To be ported over
- */
+void Utility::GetTypes(int len, std::vector<Field::Type> &types) {
+  int idx = 0;
+  while (idx < len) {
+    types.push_back(Field::Type::INTEGER);
+    idx++;
+  }
 }
+
+void Utility::GetStrings(int len, std::string val, std::vector<std::string> &strings) {
+  int idx = 0;
+  while (idx < len) {
+    strings.push_back(val + std::to_string(idx));
+    idx++;
+  }
+}
+
+TupleDesc* Utility::GetTupleDesc(int n, std::string name) {
+  std::vector<Field::Type> types;
+  std::vector<std::string> strings;
+  GetTypes(n, types);
+  GetStrings(n, name, strings);
+  return new TupleDesc(types, strings);
+}
+
+
+TupleDesc* Utility::GetTupleDesc(int n) {
+  std::vector<Field::Type> types;
+  GetTypes(n, types);
+  return new TupleDesc(types);
+}
+
+Tuple* Utility::GetHeapTuple(int n) {
+  Tuple* tup = new Tuple(*GetTupleDesc(1));
+  HeapPageId* heap_page_id = new HeapPageId(1, 2);
+  
+  tup->set_record_id(new RecordId(*heap_page_id, 3));
+  tup->set_field(0, new IntegerField(n));
+  return tup;
+}
+
+Tuple* Utility::GetHeapTuple(std::vector<int> tupdata) {
+  Tuple* tup = new Tuple(*GetTupleDesc(tupdata.size()));
+  HeapPageId* heap_page_id = new HeapPageId(1, 2);
+  
+  tup->set_record_id(new RecordId(*heap_page_id, 3));
+  for (int i = 0; i < tupdata.size(); i++) {
+    tup->set_field(i, new IntegerField(tupdata[i]));
+  }
+  return tup;
+}
+
+Tuple* Utility::GetTuple(std::vector<int> tupledata, int width) {
+  if (tupledata.size() != width) {
+    std::cout << "Get Hash Tuple has the wrong length!" << std::endl;
```

Comment:
Please throw an exception here, instead of using cout.

Do let me know if you need help with this.

There are some examples of using exceptions in tuple_desc.cpp, you can refer there if you like!